### ISSUE TITLE
20221230 support shuffle for 2d image processing

### ISSUE THREAD

**Issue by moellep** [radiasoft/sirepo#4995]
Currently the "Shuffle Before Epoch" option is ignored for image processing because the hdf5 reader does not support this. We want to avoid reading the full dataset into memory, but there may be keras data access utilities which allow loading data only for the required batch when training, which would allow for shuffling to be allowed.

**Comment by gurhar1133** [comment]
Looks like [tf.keras.preprocessing.image.ImageDataGenerator](https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/image/ImageDataGenerator) might be the way to go

From this [article](https://machinelearningmastery.com/how-to-load-large-datasets-from-directories-for-deep-learning-with-keras/):
"Instead of loading all images into memory, it will load just enough images into memory for the current and perhaps the next few mini-batches when training and evaluating a deep learning model. I refer to this as progressive loading, as the dataset is progressively loaded from file, retrieving just enough data for what is needed immediately."

Downside is that the images need to be sorted into separate train, test, and validation directories. Which maybe isn't too big of a deal

ImageDataGenerator has some functionality to handle this however. ImageDataGenerator can split train and validation if you specify the right params. So maybe we split test from the dataset, then use ImagaDataGenerator to split train and validation

**Comment by gurhar1133** [comment]
It also appears that ImageDataGenerator expects data/ to split subdirectories by label. eg.
```
data/train/
data/train/red/
data/train/blue/
data/test/
data/test/red/
data/test/blue/
data/validation/
data/validation/red/
data/validation/blue/
```
^ assumes two image categories: red and blue
